src/Ads/index.js

```diff
@@ -16,223 +16,77 @@
-    videoEl.style.visibility = 'visible'
-    videoEl.src = mediaUrl(ad.url)
-    videoEl.load()
-
-    let timeEvents = null
-    let timeout
-
-    const cleanup = () => {
-      // remove all listeners
-      Object.keys(handlers).forEach(handler =>
-        videoEl.removeEventListener(handler, handlers[handler])
-      )
-      resolve()
-    }
-    const handlers = {
-      play() {
-        Log.info('Ad', 'Play ad', ad.url)
-        fireOnConsumer('Play', ad)
-        sendBeacon(ad.callbacks, 'defaultImpression')
-      },
-      ended() {
-        fireOnConsumer('Ended', ad)
-        sendBeacon(ad.callbacks, 'complete')
-        cleanup()
-      },
-      timeupdate() {
-        if (!timeEvents && videoEl.duration) {
-          // calculate when to fire the time based events (now that duration is known)
-          timeEvents = {
-            firstQuartile: videoEl.duration / 4,
-            midPoint: videoEl.duration / 2,
-            thirdQuartile: (videoEl.duration / 4) * 3,
-          }
-          Log.info('Ad', 'Calculated quartiles times', { timeEvents })
-        }
-        if (
-          timeEvents &&
-          timeEvents.firstQuartile &&
-          videoEl.currentTime >= timeEvents.firstQuartile
-        ) {
-          fireOnConsumer('FirstQuartile', ad)
-          delete timeEvents.firstQuartile
-          sendBeacon(ad.callbacks, 'firstQuartile')
-        }
-        if (timeEvents && timeEvents.midPoint && videoEl.currentTime >= timeEvents.midPoint) {
-          fireOnConsumer('MidPoint', ad)
-          delete timeEvents.midPoint
-          sendBeacon(ad.callbacks, 'midPoint')
-        }
-        if (
-          timeEvents &&
-          timeEvents.thirdQuartile &&
-          videoEl.currentTime >= timeEvents.thirdQuartile
-        ) {
-          fireOnConsumer('ThirdQuartile', ad)
-          delete timeEvents.thirdQuartile
-          sendBeacon(ad.callbacks, 'thirdQuartile')
-        }
-      },
-      stalled() {
-        fireOnConsumer('Stalled', ad)
-        timeout = setTimeout(() => {
-          cleanup()
-        }, 5000) // make timeout configurable
-      },
-      canplay() {
-        timeout && clearTimeout(timeout)
-      },
-      error() {
-        fireOnConsumer('Error', ad)
-        cleanup()
-      },
-      // this doesn't work reliably on sky box, moved logic to timeUpdate event
-      // loadedmetadata() {
-      //   // calculate when to fire the time based events (now that duration is known)
-      //   timeEvents = {
-      //     firstQuartile: videoEl.duration / 4,
-      //     midPoint: videoEl.duration / 2,
-      //     thirdQuartile: (videoEl.duration / 4) * 3,
-      //   }
-      // },
-      abort() {
-        cleanup()
-      },
-      // todo: pause, resume, mute, unmute beacons
-    }
-    // add all listeners
-    Object.keys(handlers).forEach(handler => videoEl.addEventListener(handler, handlers[handler]))
-
-    videoEl.play()
-  })
+let getConfig = function() {
```

Comment:
We may want to make this `getAdPlatformConfig` or something, as its to expected to be used generically,.